source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1637 @@
+
+\end{code}
+
+We will now combine the postulated data above to form the least closed subset
+of B under some inductive definition ϕ and restate the closure properties and
+initiality in terms of it.
+
+\begin{code}
+
+  𝓘nd : 𝓟 {𝓤 ⊔ 𝓥 ⁺} B
+  𝓘nd b = (Ind b , Ind-trunc b)
+
+  𝓘nd-is-c-closed : c-closure 𝓘nd
+  𝓘nd-is-c-closed = c-closed
+
+  𝓘nd-is-ϕ-closed : (ϕ closure) 𝓘nd
+  𝓘nd-is-ϕ-closed = ϕ-closed
+
+  𝓘nd-is-initial : (P : 𝓟 {𝓣} B)
+                 → c-closure P
+                 → (ϕ closure) P
+                 → 𝓘nd ⊆ P
+  𝓘nd-is-initial = Ind-initial
+
+\end{code}
+
+We now consider a restricted class of inductive definitions which we will call
+local. A local inductive definition ϕ is one such that the type
+
+  Σ b ꞉ B , (Ǝ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ a)
+
+is small.
+
+We then define an operator parameterized by local inductive definitions
+and prove that it is monotone. Finally, we show that any monotone endo map on
+a sup-lattice determines a monotone operator and corresponding local
+inductive definition.
+
+\begin{code}
+
+module local-inductive-definitions
+        {𝓤 𝓦 𝓥 : Universe}
+        {B : 𝓥  ̇}
+        (L : Sup-Lattice 𝓤 𝓦 𝓥)
+        (β : B → ⟨ L ⟩)
+        (h : is-small-basis L β)
+       where
+
+ private
+  _≤_ = order-of L
+  ⋁_ = join-of L
+
+ open Joins _≤_ 
+ open is-small-basis h
+
+\end{code}
+
+We now define an auxilary subset/total space which we will use to define the
+upcoming monotone operator. This subset is in some sense a generalized downset
+that depends on ϕ.
+
+\begin{code}
+
+ _↓_ : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)) → (a : ⟨ L ⟩) → 𝓤 ⊔ 𝓦 ⊔ 𝓥  ̇
+ ϕ ↓ a = Σ b ꞉ B , (Ǝ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ a) holds) holds
+
+ ↓-to-base : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)) → (a : ⟨ L ⟩) → ϕ ↓ a → B
+ ↓-to-base ϕ a = pr₁
+
+ ↓-monotonicity-lemma : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩))
+                      → (x y : ⟨ L ⟩)
+                      → (x ≤ y) holds
+                      → ϕ ↓ x
+                      → ϕ ↓ y
+ ↓-monotonicity-lemma ϕ x y o (b , c) = (b , inclusion c)
+  where
+   inclusion : (Ǝ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ x) holds)) holds
+             → (Ǝ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ y) holds)) holds
+   inclusion = ∥∥-functor untrunc-inclusion
+    where
+     untrunc-inclusion : Σ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ x) holds)
+                       → Σ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ y) holds)
+     untrunc-inclusion (a' , p , r) = (a' , p , transitivity-of L a' x y r o)
+
+ ↓-has-sup-implies-monotone : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩))
+                            → (x y s s' : ⟨ L ⟩)
+                            → (x ≤ y) holds
+                            → (s is-lub-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds
+                            → (s' is-lub-of (ϕ ↓ y , β ∘ ↓-to-base ϕ y)) holds
+                            → (s ≤ s') holds
+ ↓-has-sup-implies-monotone
+  ϕ x y s s' o (is-upbnd , is-least-upbnd) (is-upbnd' , is-least-upbnd') =
```

Comment:
I think the convention in TypeTopology is to avoid overly descriptive names for variables such as `is-least-upbnd`, though it hasn’t been written down in `CONTRIBUTING.md`. It’s fine to merge without fixing this, but it would be good to avoid this in the future.

Reply:
Not sure this is true, see e.g. https://www.cs.bham.ac.uk/~mhe/TypeTopology/Ordinals.Injectivity.html